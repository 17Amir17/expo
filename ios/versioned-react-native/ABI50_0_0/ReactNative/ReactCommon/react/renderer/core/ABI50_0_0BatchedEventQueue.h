/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#pragma once

#include <ABI50_0_0React/renderer/core/ABI50_0_0EventQueue.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0EventQueueProcessor.h>

namespace ABI50_0_0facebook::ABI50_0_0React {

/*
 * Event Queue that dispatches event in batches synchronizing them with
 * an Event Beat.
 */
class BatchedEventQueue final : public EventQueue {
 public:
  BatchedEventQueue(
      EventQueueProcessor eventProcessor,
      std::unique_ptr<EventBeat> eventBeat);

  void onEnqueue() const override;
};

} // namespace ABI50_0_0facebook::ABI50_0_0React
