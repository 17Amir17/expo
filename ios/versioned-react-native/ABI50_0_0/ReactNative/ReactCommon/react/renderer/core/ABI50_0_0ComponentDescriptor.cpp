/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#include "ABI50_0_0ComponentDescriptor.h"

namespace ABI50_0_0facebook::ABI50_0_0React {

ComponentDescriptor::ComponentDescriptor(
    const ComponentDescriptorParameters& parameters)
    : eventDispatcher_(parameters.eventDispatcher),
      contextContainer_(parameters.contextContainer),
      flavor_(parameters.flavor) {}

const ContextContainer::Shared& ComponentDescriptor::getContextContainer()
    const {
  return contextContainer_;
}

} // namespace ABI50_0_0facebook::ABI50_0_0React
