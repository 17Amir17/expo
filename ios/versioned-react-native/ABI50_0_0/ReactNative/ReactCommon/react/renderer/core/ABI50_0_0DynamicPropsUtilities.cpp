/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#include "ABI50_0_0DynamicPropsUtilities.h"

namespace ABI50_0_0facebook::ABI50_0_0React {

folly::dynamic mergeDynamicProps(
    const folly::dynamic& source,
    const folly::dynamic& patch) {
  auto result = source;

  if (!result.isObject()) {
    result = folly::dynamic::object();
  }

  if (!patch.isObject()) {
    return result;
  }

  // Note, here we have to preserve sub-prop objects with `null` value as
  // an indication for the legacy mounting layer that it needs to clean them up.
  for (const auto& pair : patch.items()) {
    result[pair.first] = pair.second;
  }

  return result;
}

} // namespace ABI50_0_0facebook::ABI50_0_0React
