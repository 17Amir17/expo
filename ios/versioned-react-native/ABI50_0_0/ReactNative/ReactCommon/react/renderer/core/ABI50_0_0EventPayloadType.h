/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#pragma once

namespace ABI50_0_0facebook::ABI50_0_0React {

enum class EventPayloadType { ValueFactory, PointerEvent };

}
