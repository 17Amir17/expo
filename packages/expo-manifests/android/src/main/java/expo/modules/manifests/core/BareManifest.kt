package expo.modules.manifests.core

import expo.modules.jsonutils.require
import org.json.JSONException
import org.json.JSONObject

data class BareManifest(private val json: JSONObject) : BaseLegacyManifest {
  override fun getRawJson(): JSONObject {
    return json
  }

  @Deprecated(message = "Prefer to use specific field getters")
  override fun toString(): String {
    return getRawJson().toString()
  }

  /**
   * A UUID for this manifest.
   */
  @Throws(JSONException::class)
  fun getID(): String = json.require("id")

  @Throws(JSONException::class)
  fun getCommitTimeLong(): Long = json.require("commitTime")
}
