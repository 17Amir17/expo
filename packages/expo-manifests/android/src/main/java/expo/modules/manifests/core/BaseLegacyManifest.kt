package expo.modules.manifests.core

import expo.modules.jsonutils.getNullable
import expo.modules.jsonutils.require
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

interface BaseLegacyManifest : Manifest {
  override fun getStableLegacyID(): String = getRawJson().getNullable("originalFullName") ?: getLegacyID()

  override fun getScopeKey(): String = getRawJson().getNullable("scopeKey") ?: getStableLegacyID()

  override fun getEASProjectID(): String? = getRawJson().getNullable("projectId")

  override fun getAssets(): JSONArray? = getRawJson().getNullable("assets")

  @Throws(JSONException::class)
  override fun getBundleURL(): String = getRawJson().require("bundleUrl")

  override fun getExpoGoSDKVersion(): String? = getRawJson().getNullable("sdkVersion")

  override fun getExpoGoConfigRootObject(): JSONObject? {
    return getRawJson()
  }

  override fun getExpoClientConfigRootObject(): JSONObject? {
    return getRawJson()
  }

  override fun getSlug(): String? = getRawJson().getNullable("slug")

  override fun getAppKey(): String? = getRawJson().getNullable("appKey")

  fun getCommitTime(): String? = getRawJson().getNullable("commitTime")
}
