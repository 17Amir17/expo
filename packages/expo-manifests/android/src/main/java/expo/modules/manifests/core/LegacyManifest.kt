package expo.modules.manifests.core

import expo.modules.jsonutils.getNullable
import expo.modules.jsonutils.require
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

data class LegacyManifest(private val json: JSONObject) : BaseLegacyManifest {
  override fun getRawJson(): JSONObject {
    return json
  }

  @Deprecated(message = "Prefer to use specific field getters")
  override fun toString(): String {
    return getRawJson().toString()
  }

  @Throws(JSONException::class)
  fun getBundleKey(): String? = json.getNullable("bundleKey")

  @Throws(JSONException::class)
  fun getReleaseId(): String = json.require("releaseId")

  fun getRuntimeVersion(): String? = json.getNullable("runtimeVersion")

  @Throws(JSONException::class)
  fun getBundledAssets(): JSONArray? = json.getNullable("bundledAssets")

  open fun getAssetUrlOverride(): String? = json.getNullable("assetUrlOverride")
}
