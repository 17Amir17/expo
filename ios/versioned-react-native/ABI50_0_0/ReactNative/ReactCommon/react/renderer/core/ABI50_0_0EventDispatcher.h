/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#pragma once

#include <ABI50_0_0React/renderer/core/ABI50_0_0BatchedEventQueue.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0EventBeat.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0EventListener.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0EventPriority.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0EventQueueProcessor.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0StateUpdate.h>
#include <ABI50_0_0React/renderer/core/ABI50_0_0UnbatchedEventQueue.h>

namespace ABI50_0_0facebook::ABI50_0_0React {

struct RawEvent;

/*
 * Represents event-delivery infrastructure.
 * Particular `EventEmitter` clases use this for sending events.
 */
class EventDispatcher {
 public:
  using Shared = std::shared_ptr<const EventDispatcher>;
  using Weak = std::weak_ptr<const EventDispatcher>;

  EventDispatcher(
      const EventQueueProcessor& eventProcessor,
      const EventBeat::Factory& synchonousEventBeatFactory,
      const EventBeat::Factory& asynchronousEventBeatFactory,
      const EventBeat::SharedOwnerBox& ownerBox);

  /*
   * Dispatches a raw event with given priority using event-delivery pipe.
   */
  void dispatchEvent(RawEvent&& rawEvent, EventPriority priority) const;

  /*
   * Dispatches a raw event with asynchronous batched priority. Before the
   * dispatch we make sure that no other RawEvent of same type and same target
   * is on the queue.
   */
  void dispatchUniqueEvent(RawEvent&& rawEvent) const;

  /*
   * Dispatches a state update with given priority.
   */
  void dispatchStateUpdate(StateUpdate&& stateUpdate, EventPriority priority)
      const;

#pragma mark - Event listeners
  /*
   * Adds provided event listener to the event dispatcher.
   */
  void addListener(const std::shared_ptr<const EventListener>& listener) const;

  /*
   * Removes provided event listener to the event dispatcher.
   */
  void removeListener(
      const std::shared_ptr<const EventListener>& listener) const;

 private:
  const EventQueue& getEventQueue(EventPriority priority) const;

  std::unique_ptr<UnbatchedEventQueue> synchronousUnbatchedQueue_;
  std::unique_ptr<BatchedEventQueue> synchronousBatchedQueue_;
  std::unique_ptr<UnbatchedEventQueue> asynchronousUnbatchedQueue_;
  std::unique_ptr<BatchedEventQueue> asynchronousBatchedQueue_;

  mutable EventListenerContainer eventListeners_;
};

} // namespace ABI50_0_0facebook::ABI50_0_0React
