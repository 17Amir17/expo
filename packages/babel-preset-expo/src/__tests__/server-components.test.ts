import * as babel from '@babel/core';

import preset from '..';

const ENABLED_CALLER = {
  name: 'metro',
  isDev: false,
  isServer: false,
  projectRoot: '/',
};

function getCaller(props: Record<string, string | boolean>): babel.TransformCaller {
  return props as unknown as babel.TransformCaller;
}

const DEF_OPTIONS = {
  // Ensure this is absolute to prevent the filename from being converted to absolute and breaking CI tests.
  filename: '/unknown',

  babelrc: false,
  presets: [[preset, { disableImportExportTransform: true }]],
  sourceMaps: true,
  configFile: false,
  compact: false,
  comments: true,
  retainLines: false,
  caller: getCaller({ ...ENABLED_CALLER, platform: 'ios' }),
};

const originalEnv = process.env;

beforeEach(() => {
  process.env = { ...originalEnv };
});

afterAll(() => {
  process.env = { ...originalEnv };
});

it(`does nothing without use client directive`, () => {
  const options = {
    ...DEF_OPTIONS,
    caller: getCaller({ ...ENABLED_CALLER, isRSC: true, platform: 'ios' }),
  };

  const sourceCode = `
  export const foo = 'bar';
`;

  const contents = babel.transform(sourceCode, options)!.code;

  expect(contents).toMatchSnapshot();
});

it(`replaces client exports with React client references`, () => {
  const options = {
    ...DEF_OPTIONS,
    caller: getCaller({ ...ENABLED_CALLER, isRSC: true, platform: 'ios' }),
  };

  const sourceCode = `
"use client";
import { Text } from 'react-native';

export const foo = 'bar';

export default function App() {
  return <Text>Hello World</Text>
}
`;

  const contents = babel.transform(sourceCode, options)!.code;
  expect(contents).toMatchSnapshot();
});

it(`collects metadata with React client references`, () => {
  const options = {
    ...DEF_OPTIONS,
    caller: getCaller({ ...ENABLED_CALLER, isRSC: false, platform: 'ios' }),
  };

  const sourceCode = `
"use client";
import { Text } from 'react-native';

export const foo = 'bar';

export default function App() {
  return <Text>Hello World</Text>
}
`;

  const contents = babel.transform(sourceCode, options);
  expect(contents.metadata).toEqual({
    clientReferences: { entryPoint: '/unknown', exports: ['foo', 'default'] },
  });
  //   expect(contents.code).toMatchSnapshot();
});
