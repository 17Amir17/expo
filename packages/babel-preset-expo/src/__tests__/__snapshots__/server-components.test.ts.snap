// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does nothing without use client directive 1`] = `"export var foo = 'bar';"`;

exports[`replaces client exports with React client references 1`] = `
"export var foo = {
  $$typeof: "react.client.reference",
  $$async: false,
  $$id: "/unknown#foo",
  name: "foo"
};
export default {
  $$typeof: "react.client.reference",
  $$async: false,
  $$id: "/unknown#default",
  name: "default"
};"
`;
